tests/study_tests/test_optimize.py

```diff
@@ -1,49 +1,54 @@
+from unittest import mock
+
 import pytest
 
 from optuna import create_study
 from optuna import Trial
 from optuna.study import _optimize
+from optuna.study._tell import _tell_with_warning
 from optuna.testing.storage import STORAGE_MODES
 from optuna.testing.storage import StorageSupplier
 from optuna.trial import TrialState
 
 
 @pytest.mark.parametrize("storage_mode", STORAGE_MODES)
-def test_run_trial(storage_mode: str) -> None:
+def test_run_trial_catch_exception(storage_mode: str) -> None:
+    def func_value_error(_: Trial) -> float:
+        raise ValueError
 
     with StorageSupplier(storage_mode) as storage:
         study = create_study(storage=storage)
-
-        # Test trial with acceptable exception.
-        def func_value_error(_: Trial) -> float:
-
-            raise ValueError
-
-        trial = _optimize._run_trial(study, func_value_error, catch=(ValueError,))
-        frozen_trial = study._storage.get_trial(trial._trial_id)
-
+        frozen_trial = _optimize._run_trial(study, func_value_error, catch=(ValueError,))
         assert frozen_trial.state == TrialState.FAIL
 
-        # Test trial with unacceptable exception.
-        with pytest.raises(ValueError):
-            _optimize._run_trial(study, func_value_error, catch=(ArithmeticError,))
-
-        # Test trial with invalid objective value: None
-        def func_none(_: Trial) -> float:
 
-            return None  # type: ignore
-
-        trial = _optimize._run_trial(study, func_none, catch=(Exception,))
-        frozen_trial = study._storage.get_trial(trial._trial_id)
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_run_trial_exception(storage_mode: str) -> None:
+    def func_value_error(_: Trial) -> float:
+        raise ValueError
 
-        assert frozen_trial.state == TrialState.FAIL
+    with StorageSupplier(storage_mode) as storage:
+        study = create_study(storage=storage)
+        with pytest.raises(ValueError):
+            _optimize._run_trial(study, func_value_error, ())
 
-        # Test trial with invalid objective value: nan
-        def func_nan(_: Trial) -> float:
 
-            return float("nan")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_optimize_invoke_tell_with_suppressing_warning(storage_mode: str) -> None:
```

Comment:
To match the other function names.
```suggestion
def test_run_trial_invoke_tell_with_suppressing_warning(storage_mode: str) -> None:
```